### ISSUE TITLE
Xpra on Mac client connecting to Linux host doesn't trigger context change upon window changes

### ISSUE THREAD

**Issue by wolfmanstout** [talonvoice/talon#522]
[Xpra](https://xpra.org/) is an open-source X11 remote access system that allows remote windows to be forwarded and show up almost like native windows on the client (called seamless mode). On Mac, these windows will all be grouped under the Xpra application. 

To reproduce this issue, first Xpra must be installed on both the host and the client. I have tested on Linux and Mac respectively (unfortunately it looks like Mac can't act as a host, so I couldn't find an easy way to test via localhost). Mac can then start a session on Linux with: `xpra start ssh://USERNAME@HOST/ --start=xterm`. From here, you can start another X11 application, e.g. Emacs in my use case. Assuming the bug reproduces, you won't see the window title context update in Talon when this new application is started and becomes the active window. The context also will not update if you do cmd-backtick. You can trigger an update by doing cmd-tab twice to switch back and forth from another app. 

I'm seeing this on MacOS 12.5 (Monterey). It's an annoying issue because I use this to remotely access my primary workstation and regularly switch between Emacs and gnome-terminal.

**Comment by nriley** [comment]
I have similar issues with Citrix, eg. see #315 — this affects not just the auth window but also individual Citrix windows.

Assuming this is the same issue, here's one workaround — if you focus another app and then back.

https://github.com/nriley/knausj_talon/blob/00739fa323cbb351907522042108d78086f54c32/apps/citrix/citrix_auth.py#L17

Or another one when I'm trying to wait for windows to appear — I poll for windows with accessibility, then try to get Talon to notice the new window when it appears.

```python
        # XXX workaround for Talon not noticing Citrix windows
        self.window_titles = set()
        self.window_wait_job = cron.interval("500ms", self.look_for_windows)

    def look_for_windows(self):
        # XXX work around Talon not noticing new windows
        app = ui.apps(pid=self.pid)
        if not app:
            cron.cancel(self.window_wait_job)
            return

        app = app[0]
        windows = app.children.find(AXRole="AXWindow", max_depth=0)
        window_titles = set(window.AXTitle for window in windows)
        if window_titles == self.window_titles:
            return

        print("Talon thinks Citrix session windows are", app.windows())
        print("Citrix session windows are", windows)

        for window in windows:
            title = window.AXTitle
            if title in self.window_titles:
                continue
            if title.startswith("...") or title.startswith("..."):
                print("Attempting to make Talon notice Citrix is focused")
                focus_window(window, app)
                cron.cancel(self.window_wait_job)
                return
            elif title.startswith("..."):
                print(
                    "Attempting to make Talon notice [...] is focused"
                )
                focus_window(window, app)
                break

        self.window_titles = window_titles

def focus_window(window, app):
    ui.apps(pid=os.getpid())[0].focus()
    app.focus()
    window.perform("AXRaise")
```

**Comment by nriley** [comment]
This may be fixed now in beta (see #502) — at least I am able to remove my workarounds.

**Comment by lunixbochs** [comment]
@wolfmanstout I believe this is fixed as of the #502 fixes, please re-open if not.

**Comment by wolfmanstout** [comment]
I tried testing this earlier but Xpra is broken on my work Mac so I wasn't able verify. I'm no longer reliant on this, though.
